cleo_plugins/FileSystemOperations/FileUtils.cpp

```diff
@@ -18,465 +18,492 @@ DWORD File::FUNC_ferror = 0;
 {
-	if (file == nullptr) return 0;
+    if (file == nullptr)
+        return 0;
 
-	auto handle = (DWORD)file;
-	if (!legacy) handle |= 0x1;
-	return handle;
+    auto handle = (DWORD)file;
+    if (!legacy)
+        handle |= 0x1;
+    return handle;
 }
 
 void File::updateState(DWORD handle)
 {
-	// RW streams needs synchronization of read and write positions (https://en.wikibooks.org/wiki/C_Programming/stdio.h/fopen)
-	if (isOk(handle) && !isEndOfFile(handle))
-	{
-		seek(handle, 0, SEEK_CUR);
-	}
+    // RW streams needs synchronization of read and write positions
+    // (https://en.wikibooks.org/wiki/C_Programming/stdio.h/fopen)
+    if (isOk(handle) && !isEndOfFile(handle))
+    {
+        seek(handle, 0, SEEK_CUR);
+    }
 }
 
 bool File::flush(DWORD handle)
 {
-	FILE* file = handleToFile(handle);
-	if (file == nullptr) return false;
-
-	int result = 0;
-	if (isLegacy(handle))
-	{
-		_asm
-		{
+    FILE* file = handleToFile(handle);
+    if (file == nullptr)
+        return false;
+
+    int result = 0;
+    if (isLegacy(handle))
+    {
+        _asm
+        {
 			push file
 			call FUNC_fflush
 			add esp, 0x4
 			mov result, eax
-		}
-	}
-	else
-		result = fflush(file);
+        }
+    }
+    else
+        result = fflush(file);
 
-	return result == 0;
+    return result == 0;
 }
 
 DWORD File::open(const char* filename, const char* mode, bool legacy)
 {
-	// validate the mode argument
-	if (!legacy)
-	{
-		static char modeUpdated[12];
-		const std::string allowed = "+abcnrtwxDRST"; // https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170
-
-		bool valid = false;
-		bool binary = false;
-		bool text = false;
-		auto modeLen = mode != nullptr ? strlen(mode) : 0;
-		if (modeLen > 0 && modeLen < (sizeof(modeUpdated) - 1)) // keep space for extra binary mode char
-		{
-			valid = true;
-
-			for (auto ch : std::string_view(mode))
-			{
-				if (allowed.find(ch) == std::string_view::npos)
-				{
-					valid = false;
-					break; // invalid character
-				}
-
-				if (ch == 'b') binary = true;
-				if (ch == 't') text = true;
-			}
-
-			if (binary && text) valid = false;
-
-			// By default open as binary mode.
-			// Generally text mode is not well documented in C and many file related functions has undefined behavior. For example 'ftell' returns invalid values.
-			if (valid && (!binary && !text))
-			{
-				strcpy_s(modeUpdated, mode);
-				strcat_s(modeUpdated, "b");
-				mode = modeUpdated;
-			}
-		}
-
-		if (!valid)
-		{
-			LOG_WARNING(0, "Invalid mode argument '%s' while opening file \"%s\" stream!", mode, filename);
-			return 0; // invalid handle
-		}
-	}
-
-	FILE* file = nullptr;
-	if (legacy)
-	{
-		_asm
-		{
+    // validate the mode argument
+    if (!legacy)
+    {
+        static char modeUpdated[12];
+        const std::string allowed =
+            "+abcnrtwxDRST"; // https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170
```

Comment:
so apparently long comments force code to be pushed to the next line. I think in this case it's better to move comment at the top:

```
// https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170
const std::string allowed = "+abcnrtwxDRST";
```